symengine/pow.cpp

```diff
@@ -164,6 +183,39 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
         RCP<const Pow> A = rcp_static_cast<const Pow>(a);
         return pow(A->get_base(), mul(A->get_exp(), b));
     }
+    if (eq(*a, *E)) {
+        if (is_a<Mul>(*b) and exp_mul_helper(b)) {
+            RCP<const Number> coef = down_cast<const ComplexBase &>(
+                                         *down_cast<const Mul &>(*b).get_coef())
+                                         .imaginary_part();
+            RCP<const Basic> arg = mul(coef, integer(2));
+            long n = ((down_cast<const Integer &>(*arg).as_int() % 4) + 4) % 4;
+            if (!n) {
+                return one;
+            } else if (n == 1) {
+                return I;
+            } else if (n == 2) {
+                return minus_one;
+            } else {
+                return Complex::from_two_nums(*zero, *minus_one);
+            }
+        } else if (is_a<Add>(*b)) {
+            const umap_basic_num &dict = down_cast<const Add &>(*b).get_dict();
+            umap_basic_num new_dict;
+            RCP<const Number> coef = down_cast<const Add &>(*b).get_coef();
+            RCP<const Basic> s = one;
+            for (const auto &p : dict) {
+                if (eq(*p.first, *pi)
```

Comment:
To do this, you don't need to iterate the dictionary. Just use `dict.find` to find the value `pi`.

Reply:
Is there a way I can remove that particular entry of `pi` as well? That way, there'd be no need of iteration at all.